src/Domain/ElementMap.cpp

```diff
@@ -0,0 +1,67 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/ElementMap.hpp"
+
+/// \cond
+template <size_t Dim, typename TargetFrame>
+ElementMap<Dim, TargetFrame>::ElementMap(
+    ElementId<Dim> element_id,
+    std::unique_ptr<CoordinateMapBase<Frame::Logical, TargetFrame, Dim>>
+        block_map) noexcept
+    : block_map_(std::move(block_map)),
+      element_id_(std::move(element_id)),
+      length_of_range_{[](const ElementId<Dim>& id) {
+        std::array<double, Dim> result{};
+        for (size_t d = 0; d < Dim; ++d) {
+          gsl::at(result, d) =
+              gsl::at(id.segment_ids(), d).endpoint(Side::Upper) -
+              gsl::at(id.segment_ids(), d).endpoint(Side::Lower);
+        }
+        return result;
+      }(element_id_)},
+      sum_of_endpoints_{[](const ElementId<Dim>& id) {
+        std::array<double, Dim> result{};
+        for (size_t d = 0; d < Dim; ++d) {
+          gsl::at(result, d) =
+              gsl::at(id.segment_ids(), d).endpoint(Side::Upper) +
+              gsl::at(id.segment_ids(), d).endpoint(Side::Lower);
+        }
+        return result;
+      }(element_id_)},
+      map_slope_{0.5 * length_of_range_},
+      map_offset_{0.5 * sum_of_endpoints_},
+      map_inverse_slope_{[this]() {
+        std::array<double, Dim> result{};
+        for (size_t d = 0; d < Dim; ++d) {
+          gsl::at(result, d) = 2.0 / gsl::at(this->length_of_range_, d);
+        }
+        return result;
+      }()},
+      map_inverse_offset_{[this]() {
+        std::array<double, Dim> result{};
+        for (size_t d = 0; d < Dim; ++d) {
+          gsl::at(result, d) = -gsl::at(this->sum_of_endpoints_, d) /
+                               gsl::at(this->length_of_range_, d);
+        }
+        return result;
+      }()},
+      jacobian_{0.5 * length_of_range_},
+      inverse_jacobian_{map_inverse_slope_} {}
+
+template <size_t Dim, typename TargetFrame>
+void ElementMap<Dim, TargetFrame>::pup(PUP::er& p) noexcept {
+  p | block_map_;
+  p | element_id_;
+  p | map_slope_;
```

Comment:
`length_of_range_` and `sum_of_endpoints_` are missing.  They are also unused, except during initialization.  Can we not store them?

Reply:
Sure. Do you think adding a function to compute them that is called in the lambdas, or moving the initialization of everything that follows to the constructor body is better?

Reply:
How about just using `2. * map_slope_` for `length_of_range_` and `2. * map_offset_` for `sum_of_endpoints_`?